lib/msal-common/src/network/ThrottlingManager.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, NetworkRequestOptions } from "./INetworkModule";
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { Constants, HeaderNames } from "../utils/Constants";
+import { RequestThumbprint } from "./RequestThumbprint";
+import { RequestThumbprintValue } from "./RequestThumbprintValue";
+import { AuthError } from "../error/AuthError";
+
+export type NetworkResponse<T> = {
+    headers: Map<string, string>;
+    body: T;
+    status: number;
+};
+
+export abstract class ThrottlingManager {
```

Comment:
Not sure I like calling this `ThrottlingManager`, would rather keep it as `NetworkManager`. IMO throttling is an implementation detail of our network class.

Reply:
+1 I think `NetworkManager` should stay as is and call the throttling functions when needed

Reply:
@tnorling what do you consider "throttling functions"? the ones that are currently abstract?

Reply:
Anything that is for the purpose of throttling, e.g. `preProcess`, `postProcess`, `getItem`, `setItem`,etc.

Reply:
So then the point of `NetworkManager` existing at all would be exclusively to hook into pre/post API call? Because at that point, the only method in the class is `sendPostRequest`. I kind of feel like it can be combined with what's in `INetworkModule` at that point, if I'm understanding correctly

Reply:
Yeah, I suppose since the `NetworkManager` isn't implemented today you could just put the calls to `preProcess` and `postProcess` in the library implementations of the `INetworkModule`

Reply:
I've kept `NetworkManager` separate for now as to not change the INetworkModule interface and to keep the implementation in common.